rhoas/kafkas/resource_kafka.go

```diff
@@ -95,12 +95,12 @@ func kafkaDelete(ctx context.Context, d *schema.ResourceData, m interface{}) dia
 	// Warning or errors can be collected in a slice type
 	var diags diag.Diagnostics
 
-	c, ok := m.(*rhoasClients.Clients)
+	api, ok := m.(rhoasAPI.API)
```

Comment:
Slightly nit picky, but you've renamed the `clients` to `api`. I would have left them as is, as each client points to the respective API i.e. kafkaMgmt client -> kafkaMgmt API. I feel like its clearer but thats my take. However, you've not changed the error messages to reflect. they still refer to clients. This may lead to some confusion but I'll leave that up to you.